components/create-employee-modal.tsx

```diff
@@ -0,0 +1,124 @@
+'use client';
+import { useState, useEffect } from 'react';
+import { Modal } from './Modal';
+import { CreateEmployeeModalProps } from '@/models/global-types';
+import { Input } from '@/components/input';
+import { Button } from '@/components/button';
+import { ExIcon } from '@/assets/icons';
+
+const CreateEmployeeModal = ({
+  modalIsOpen,
+  setModalIsOpen = () => {},
+  formData,
+  setFormData = () => {},
+  formErrors,
+  setFormErrors = () => {},
+}: CreateEmployeeModalProps) => {
+  const closeModal = () => {
+    setModalIsOpen(false);
+  };
+
+  const handleInputChange = (e: any) => {
+    const { name, value } = e.target;
+
+    setFormData((prev: any) => {
+      return { ...prev, [name]: value };
+    });
+
+    setFormErrors((prev: any) => {
+      return { ...prev, [name]: '' };
+    });
+  };
+
+  useEffect(() => {
```

Comment:
I hope we can move this `useEffect` at the top level of this component, i.e. we can put it above `closeModal` handler.